SEPADebit/src/test/java/com/braintreepayments/api/SEPADebitApiUnitTest.java

```diff
@@ -0,0 +1,187 @@
+package com.braintreepayments.api;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mock;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.robolectric.RobolectricTestRunner;
+
+@RunWith(RobolectricTestRunner.class)
+public class SEPADebitApiUnitTest {
+
+    private HttpClient httpClient;
+    private CreateMandateCallback createMandateCallback;
+
+    @Before
+    public void beforeEach() {
+        httpClient = mock(HttpClient.class);
+        createMandateCallback = mock(CreateMandateCallback.class);
+    }
+
+    @Test
+    public void createMandate_onSuccessfulHttpResponse_callsBackCreateMandateResult() {
+        doAnswer(new Answer<Void>() {
```

Comment:
We typically move `doAnswer` mock code into a builder class i.e. `MockSEPADebitApiBuilder` that produces a mock with the fixtures configured. One example is [MockBraintreeClientBuilder.java](https://github.com/braintree/braintree_android/blob/master/TestUtils/src/main/java/com/braintreepayments/api/MockBraintreeClientBuilder.java), the sepa one should be much smaller though.

Reply:
Yeah for sure - we are working on another PR that will add a `MockSEPADebitApiBuilder` when we test these flows in the client. For this PR, we didn't create a mock builder because the HttpClient will eventually be updated to an ApiClient when this feature is ready in Sandbox (we have to use http for now to directly hit the cpair gateway).